packages/xsnap/test/xsnap-eof.test.js

```diff
@@ -0,0 +1,268 @@
+/* global setTimeout */
+
+import test from 'ava';
+import * as proc from 'child_process';
+import * as os from 'os';
+import fs from 'fs';
+import process from 'node:process';
+import { tmpName } from 'tmp';
+import { xsnap } from '../src/xsnap.js';
+import { options } from './message-tools.js';
+
+const sleep = ms => new Promise(res => setTimeout(res, ms));
+
+async function waitFor(conditionFn) {
+  while (!conditionFn()) {
+    await sleep(100);
+  }
+}
+
+test('xsnap-worker complains while waiting for answer when parent is killed', async t => {
+  const parentText = `
+        import * as proc from 'child_process';
+        import * as os from 'os';
+        import fs from 'fs';
+        import process from 'node:process';
+        import { tmpName } from 'tmp';
+        import '@endo/init';
+        import { xsnap } from '../src/xsnap.js';
+        import { options, decode, encode, loader } from './message-tools.js';
+        import console from 'console';
+        const io = { spawn: proc.spawn, os: os.type(), fs, tmpName };
+
+        async function handleCommand(message) {
+            process.kill(process.pid, 'SIGKILL');
+            return new Uint8Array();
+        }
+
+        const vat = await xsnap({ ...options(io), handleCommand });
+        await vat.evaluate('function handleCommand(message) { issueCommand(new Uint8Array().buffer); };');
+        await vat.issueStringCommand('0');
+        console.error('Error: parent was not killed!');
+        await vat.close();
+    `;
+
+  let nodeExited = false;
+  let nodeError = null;
+  let nodeExitCode = null;
+  let nodeExitSignal = null;
+  let strOut = '';
+  let strErr = '';
+
+  const p = proc.spawn('node', ['--input-type=module', '-e', parentText], {
+    cwd: `${process.cwd()}/test`,
+  });
+
+  p.stdout.on('data', data => {
+    strOut = `${strOut}${data}`;
+  });
+
+  p.stderr.on('data', data => {
+    strErr = `${strErr}${data}`;
+  });
+
+  p.on('error', err => {
+    nodeError = err;
+  });
+
+  p.on('exit', (code, signal) => {
+    nodeExited = true;
+    nodeExitCode = code;
+    nodeExitSignal = signal;
+  });
+
+  p.stdin.end();
+
+  await waitFor(() => {
+    return nodeExited;
+  });
+
+  t.is(`${strOut}`, ``, 'wrong stdout output');
+  t.regex(`${strErr}`, /Has parent died\?/, 'wrong stderr output');
+  t.is(nodeError, null, 'parent returned an error');
+  t.is(nodeExited, true, 'parent did not exit');
+  t.is(nodeExitCode, null, 'parent exit code is not null');
+  t.is(`${nodeExitSignal}`, `SIGKILL`, 'wrong parent exit signal');
```

Comment:
There's no need for these to be template literals.
```suggestion
  t.is(strOut, '', 'wrong stdout output');
  t.regex(strErr, /Has parent died\?/, 'wrong stderr output');
  t.is(nodeError, null, 'parent returned an error');
  t.is(nodeExited, true, 'parent did not exit');
  t.is(nodeExitCode, null, 'parent exit code is not null');
  t.is(nodeExitSignal, `SIGKILL`, 'wrong parent exit signal');
```